source/core/simulation_control.cc

```diff
@@ -48,6 +55,116 @@ SimulationControl::is_output_iteration()
+    }
+  else if (iteration_number == 3 &&
+           method == Parameters::SimulationControl::TimeSteppingMethod::bdf2 &&
+           bdf_start_method == Parameters::SimulationControl::
+                                 BDFStartupMethods::multiple_step_bdf)
+    {
+      assembly_method = Parameters::SimulationControl::TimeSteppingMethod::bdf2;
+      set_suggested_time_step(initial_time_step);
+    }
+  else if (iteration_number <= 1 &&
+           method == Parameters::SimulationControl::TimeSteppingMethod::bdf3 &&
+           bdf_start_method == Parameters::SimulationControl::
+                                 BDFStartupMethods::multiple_step_bdf)
+    {
+      assembly_method = Parameters::SimulationControl::TimeSteppingMethod::bdf1;
+      set_current_time_step(initial_time_step * startup_timestep_scaling);
+    }
+  else if (iteration_number == 2 &&
+           method == Parameters::SimulationControl::TimeSteppingMethod::bdf3 &&
+           bdf_start_method == Parameters::SimulationControl::
+                                 BDFStartupMethods::multiple_step_bdf)
+    {
+      assembly_method = Parameters::SimulationControl::TimeSteppingMethod::bdf2;
+      set_suggested_time_step(initial_time_step * startup_timestep_scaling);
+    }
+  else if (iteration_number == 3 &&
+           method == Parameters::SimulationControl::TimeSteppingMethod::bdf3 &&
+           bdf_start_method == Parameters::SimulationControl::
+                                 BDFStartupMethods::multiple_step_bdf)
+    {
+      assembly_method = Parameters::SimulationControl::TimeSteppingMethod::bdf3;
+      set_suggested_time_step(initial_time_step *
+                              (1 - 2 * startup_timestep_scaling));
+    }
+  else if (iteration_number == 4 &&
+           method == Parameters::SimulationControl::TimeSteppingMethod::bdf3 &&
+           bdf_start_method == Parameters::SimulationControl::
+                                 BDFStartupMethods::multiple_step_bdf)
+    {
+      assembly_method = Parameters::SimulationControl::TimeSteppingMethod::bdf3;
+      set_suggested_time_step(initial_time_step);
+    }
+  else if (iteration_number <= 1 &&
+           (method ==
+            Parameters::SimulationControl::TimeSteppingMethod::bdf2) &&
+           bdf_start_method ==
+             Parameters::SimulationControl::BDFStartupMethods::sdirk_step)
+    {
+      assembly_method =
+        Parameters::SimulationControl::TimeSteppingMethod::sdirk22;
+      set_suggested_time_step(initial_time_step);
+    }
+  else if (iteration_number <= 2 &&
+           (method ==
+            Parameters::SimulationControl::TimeSteppingMethod::bdf3) &&
+           bdf_start_method ==
+             Parameters::SimulationControl::BDFStartupMethods::sdirk_step)
+    {
+      assembly_method =
+        Parameters::SimulationControl::TimeSteppingMethod::sdirk22;
+      set_suggested_time_step(initial_time_step);
+    }
+  else
+    {
+      assembly_method = method;
+    }
```

Comment:
I think cascading "if"s would improve the readability, for example with level 1 - BDFStartupMethods, level 2 - TimeSteppingMethod, level 3 - iteration number.

Reply:
This would lead to duplication of the else statement so for the moment i think it would let it as it is for the time being.